python/lsst/analysis/drp/makeQaTractTables.py

```diff
@@ -0,0 +1,353 @@
+            ``band``.
+        """
+        # TODO: remove the filterMap config and dance once DM-28479 lands
+        if self.config.filterMap is not None:
+            if band in self.config.filterMap:
+                filterName = self.config.filterMap[band]
+            else:
+                raise RuntimeError("A config.filterMap was provided, but there is no entry for the "
+                                   "band requested: {}".format(band))
+        else:
+            filterName = band
+        self.log.info("Aggregating QA table for {} tract {} including {} patches".
+                      format(filterName, tract, len(inputObjCats)))
+        # It is much faster to concatenate a list of DataFrames than to
+        # concatenate successively within the for loop.
+        catList = []
+        for inputCat in inputObjCats:
+            patch = inputCat.dataId["patch"]
+            inputCatCols = inputCat.get(component="columns")
+            inputCatFilters = inputCatCols.get_level_values("filter").unique()
+            inputCat = inputCat.get()  # Required because deferLoad=True was set
+            haveFilter = True
+            activeFilterName = filterName
+            if filterName not in inputCatFilters:
+                patchId = inputCat["meas"][inputCatFilters[0]]["patchId"].unique()[0]
+                self.log.info("Filter {} does not exist for: {}, {:2d} [{}].  Setting NaNs for patch "
+                              "columns...".format(filterName, tract, int(patch), patchId))
+                haveFilter = False
+                activeFilterName = inputCatFilters[0]
+            cat = self.collateTable(inputCat, activeFilterName, haveFilter)
+            catList.append(cat)
+
+        self.log.info("Concatenating tables from {} patches".format(len(catList)))
+        if not catList:
+            raise RuntimeError("No catalogs read for {}.".format(inputObjCats))
+        allCats = pd.concat(catList, axis=0)
+        # The object "id" is associated with the dataframe index.  Add a
+        # column that is the id so that it is available for operations on it,
+        # e.g. cat["id"].
+        allCats["id"] = allCats.index
+        # Reindex to match sorting of objTable_tract tables
+        inputObjectTable = inputObjectTable.get(parameters={"columns": []})
+        allCats = allCats.reindex(inputObjectTable.index)
+        return pipeBase.Struct(qaTractTable=allCats)
+
+
+class MakeForcedQaTractTablesTaskConnections(BaseMakeQaTractTablesTaskConnections,
+                                             defaultTemplates={"coaddName": "deep"},
+                                             dimensions=("tract", "skymap")):
+    qaTractTable = cT.Output(
+        doc="The collated catalog of forced measurements with additional columns for QA added",
+        storageClass="DataFrame",
+        name="qaTractTable_forced",
+        dimensions=("tract", "band", "skymap"),
+    )
```

Comment:
I've been wondering whether to recommend moving the output connection to the base class, and I think the fact that the base class's `run` clearly expects it to exist settles it.

More specifically, I think we should:
 - move the `qaTractTable = cT.Output(...)` from the `MakeForced` connections class to the base one;
 - in `MakeUnforcedQaTractTablesTaskConfig`, add a `setDefaults` with `self.connections.qaTractTable = "qaTractTable_unforced"`.